src/self_driving/planning/seq_tuning/graph_solver.cpp

```diff
@@ -0,0 +1,146 @@
+#include "self_driving/planning/seq_tuning/graph_solver.h"
+
+#include <cmath>
+
+#include "loggers/selfdriving_logger.h"
+#include "self_driving/forecasting/workload_forecast.h"
+#include "self_driving/planning/action/abstract_action.h"
+#include "self_driving/planning/pilot.h"
+#include "self_driving/planning/pilot_util.h"
+#include "self_driving/planning/seq_tuning/path_solution.h"
+
+namespace noisepage::selfdriving::pilot {
+
+GraphSolver::GraphSolver(const PlanningContext &planning_context,
+                         common::ManagedPointer<selfdriving::WorkloadForecast> forecast, uint64_t end_segment_index,
+                         const std::map<action_id_t, std::unique_ptr<AbstractAction>> &structure_map,
+                         const std::vector<double> &default_segment_cost,
+                         const std::vector<std::set<std::set<action_id_t>>> &candidate_configurations_by_segment,
+                         uint64_t memory_constraint) {
+  // build the graph and compute shortest path simultaneously
+
+  ActionState action_state;
+  action_state.SetIntervals(0, end_segment_index);
```

Comment:
I don't think it's really meaning for to set intervals for the `ActionState`, because all you need the `ActionState` for is to calculate the memory consumption. I think we can just document this and remove all the `SetIntervals`.

Reply:
Since the start and end intervals are relevant when computing the hashes of an action state, would we need them for caching the result of computing memory consumption later? Or do we skip the cache in the baseline?

Reply:
I don't think we need to add caching to the baseline since they don't have that. The memory computation here is not expensive anyway since it doesn't invoke the models (we invoke models when we create actions using `PilotUtil::EstimateCreateIndexAction`).